src/img_boss/build_ks.py

```diff
@@ -832,11 +834,14 @@ def handle_lifecycle_control(self, ctrl):
                 self.oscrc = ctrl.config.get("obs", "oscrc")
 
     def get_repositories(self, wid, project):
-        if not wid.fields.ev or not wid.fields.ev.namespace:
+        if not (wid.fields.ev and wid.fields.ev.namespace):
             raise RuntimeError("Missing field: ev.namespace")
-        obs = BuildService(oscrc=self.oscrc, apiurl=wid.fields.ev.namespace)
+        if self.obs is None or self.namespace != wid.fields.ev.namespace:
```

Comment:
so if wid.fields.ev.namespace is not present due to not being an OBS event then won't "or self.namespace != wid.fields.ev.namespace"  cause a cache miss?

I see the line before makes ev.namespace mandatory ... yet commit message says "Considered also using The BuildServiceParticipant mixin, but that would always require the ev.namespace field, which is only set on processes initiated from OBS events."